upload-api/tables/index.js

```diff
@@ -26,3 +26,55 @@ export const uploadTableProps = {
   // space + root must be unique to satisfy index constraint
   primaryIndex: { partitionKey: 'space', sortKey: 'root' },
 }
+
+/** @type TableProps */
+export const delegationTableProps = {
+  fields: {
+    link: 'string',        // `baf...x`
+    audience: 'string',   // `did:web:service`
+    issuer: 'string',     // `did:key:agent`
+    expiration: 'string', // `9256939505` (unix timestamp)
+    insertedAt: 'string', // `2022-12-24T...`
+    updatedAt: 'string',  // `2022-12-24T...`
+  },
+  // TODO does this index setup seem right?
+  // we want to query by audience, but that won't necessarily be unique, so use cid as sortKey
+  primaryIndex: { partitionKey: 'audience', sortKey: 'link' },
+}
+
+/** @type TableProps */
+export const subscriptionTableProps = {
+  fields: {
+    cause: 'string',        // `baf...x` (CID of invocation that created this subscription)
+    provider: 'string',   // `did:web:service` (DID of the provider, e.g. a storage provider)
+    customer: 'string',   // `did:mailto:agent` (DID of the user account)
+    subscription: 'string',      // string (arbitrary string associated with this subscription)
+    insertedAt: 'string', // `2022-12-24T...`
+    updatedAt: 'string',  // `2022-12-24T...`
+  },
+  // TODO does this index setup seem right?
+  primaryIndex: { partitionKey: 'customer', sortKey: 'subscription' },
+  globalIndexes: {
+    // TODO: I don't think we should keep this index - partitioning by provider is basically useless and won't be much faster than a table scan
+    provider: { partitionKey: 'provider', projection: ['customer'] },
+    subscription: { partitionKey: 'subscription', projection: ['customer'] }
+
+  }
+}
+
+/** @type TableProps */
+export const consumerTableProps = {
+  fields: {
+    cause: 'string',        // `baf...x` (CID of invocation that created this consumer record)
+    consumer: 'string',   // `did:key:space` (DID of the actor that is consuming the provider, e.g. a space DID)
+    provider: 'string',   // `did:web:service` (DID of the provider, e.g. a storage provider)
+    subscription: 'string',      // string (arbitrary string associated with this subscription)
```

Comment:
Assuming we integrate `order` here I would expect this field to be `order` as opposed to `subsscription` since later could be derived from `(provider, order)` tuple.